### PR TITLE
Update `CircleAvatar` to support Material 3

### LINKED ISSUE #114811: Update `CircleAvatar` to support Material 3
image

Specs: https://m3.material.io/components/lists/specs

related https://github.com/flutter/flutter/issues/114006

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/114811

This PR updates `CircleAvatar` to support Material 3.

Screenshot 2022-11-07 at 15 31 08

Since there is no theme for the `CircleAvatar`, I just updated the default colors. 
M3 Avatar size and font are the same as the current one. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/circle_avatar.dart [hunk 1]
```diff
@@ -192,8 +195,11 @@ class CircleAvatar extends StatelessWidget {
   Widget build(BuildContext context) {
     assert(debugCheckHasMediaQuery(context));
     final ThemeData theme = Theme.of(context);
-    TextStyle textStyle = theme.primaryTextTheme.titleMedium!.copyWith(color: foregroundColor);
-    Color? effectiveBackgroundColor = backgroundColor;
+    final Color? effectiveForegroundColor = foregroundColor
+      ?? (theme.useMaterial3 ? theme.colorScheme.onPrimaryContainer : null);
+    TextStyle textStyle = theme.primaryTextTheme.titleMedium!.copyWith(color: effectiveForegroundColor);
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/circle_avatar.dart:200]: We should be using `theme.textTheme` here. primaryTextTheme, like primaryColorLight and primaryColorDark, are obsolete and will be deprecated when Material3 becomes the default.

File: packages/flutter/test/material/circle_avatar_test.dart [hunk 2]
```diff
@@ -306,6 +278,61 @@ void main() {
     final RenderParagraph paragraph = tester.renderObject(find.text('Z'));
     expect(paragraph.text.style!.color, equals(ThemeData.fallback().primaryColorLight));
   });
+
+  group('Material 2', () {
+    // Tests that are only relevant for Material 2. Once ThemeData.useMaterial3
```
Comment (HansMuller) [inline packages/flutter/test/material/circle_avatar_test.dart:283]: NICE

**Review by HansMuller — COMMENTED** [review]
This looks good, just needs one small change.

File: packages/flutter/lib/src/material/circle_avatar.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/lib/src/material/circle_avatar.dart:200]: This seems to be the M2 default color. I will look into it.


Screenshot 2022-11-08 at 18 43 39


Screenshot 2022-11-08 at 18 43 26

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**